README.md

```diff
@@ -82,17 +82,39 @@ executed.
 [Scripted](https://www.scala-sbt.org/1.x/docs/Testing-sbt-plugins.html) is a tool that allow you to test sbt plugins.
 For each test it is necessary to create a specially crafted project. These projects are inside src/sbt-test directory.
 
-Scripted tests are run using `scripted` comand.
+Scripted tests are run using `scripted` command.
 
-### scalafix
+### Formatting
+
+The codebase is formatted with [scalafmt](https://scalameta.org/scalafmt/), as such the codebase needs to be formatted
+before submitting a PR.
+
+Various runners for Scalafmt exist, such as
+* A [sbt scalafmt plugin](https://github.com/scalameta/sbt-scalafmt) that lets you run scalafmt directly within sbt using
+    * `scalafmt` to format base scala sources
+    * `test:scalafmt` to format test scala sources
+    * `scalafmtSbt` to format the `build.sbt` file
+    * `scalafmtAll` to format everything
+* IntelliJ IDEA and VSCode will automatically detect projects with scalafmt and prompt you whether to use Scalafmt. See
+  the [scalafmt installation guide][scalafmt-installation-link] for more details
+* There are native builds of Scalafmt that let you run a `scalafmt` as a CLI tool, see the CLI section in
+  [scalafmt installation guide][scalafmt-installation-link]
+
+Note that a GitHub action exists which will check that your code is formatted whenever you create a PR. For more details
+read the [documentation](https://aiven.github.io/guardian-for-apache-kafka/ci.html#scalafmt)
+
+### Linting
 
 This project uses [scalafix](https://scalacenter.github.io/scalafix/) as a linter/style guide enforcer. To run scalafix
 you can simply do
 
 ```sbt
-scalafixAll
+clean test/clean scalafixAll
```

Comment:
Since scalafix relies on the scalac compiler it only works properly if it compiles all of the code, hence why you should `clean` beforehand.